src/Options.js

```diff
@@ -59,7 +85,7 @@ class Options {
      * Returns a new options object with the same properties as "this".  Properties
      * from "extension" will be copied to the new options object.
      */
-    extend(extension) {
+    extend(extension: $Shape<OptionsData>): Options {
```

Comment:
The `$Shape` is no longer necessary.

Reply:
I'm pretty sure it's necessary. Taking it out leads to the following Flow errors:

```
Error: Options.js:116
                        v------------
116:             return this.extend({
117:                 style: style,
118:                 size: sizeAtStyle(this.textSize, style),
119:             });
                 -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                    v
116:             return this.extend({
117:                 style: style,
118:                 size: sizeAtStyle(this.textSize, style),
119:             });
                 ^ object literal

Error: Options.js:139
                        v------------
139:             return this.extend({
140:                 style: this.style.text(),
141:                 size: size,
142:                 textSize: size,
143:             });
                 -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                    v
139:             return this.extend({
140:                 style: this.style.text(),
141:                 size: size,
142:                 textSize: size,
143:             });
                 ^ object literal

Error: Options.js:158
                        v------------
158:             return this.extend({
159:                 style: style,
160:                 size: wantSize,
161:             });
                 -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                    v
158:             return this.extend({
159:                 style: style,
160:                 size: wantSize,
161:             });
                 ^ object literal

Error: Options.js:169
                    v------------
169:         return this.extend({
170:             color: color,
171:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                v
169:         return this.extend({
170:             color: color,
171:         });
             ^ object literal

Error: Options.js:169
                    v------------
169:         return this.extend({
170:             color: color,
171:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `style`. Property not found in
                                v
169:         return this.extend({
170:             color: color,
171:         });
             ^ object literal

Error: Options.js:178
                    v------------
178:         return this.extend({
179:             phantom: true,
180:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                v
178:         return this.extend({
179:             phantom: true,
180:         });
             ^ object literal

Error: Options.js:178
                    v------------
178:         return this.extend({
179:             phantom: true,
180:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `style`. Property not found in
                                v
178:         return this.extend({
179:             phantom: true,
180:         });
             ^ object literal

Error: Options.js:187
                    v------------
187:         return this.extend({
188:             font: font || this.font,
189:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `maxSize`. Property not found in
                                v
187:         return this.extend({
188:             font: font || this.font,
189:         });
             ^ object literal

Error: Options.js:187
                    v------------
187:         return this.extend({
188:             font: font || this.font,
189:         });
             -^ call of method `extend`
 88:     extend(extension: OptionsData): Options {
                           ^^^^^^^^^^^ property `style`. Property not found in
                                v
187:         return this.extend({
188:             font: font || this.font,
189:         });
             ^ object literal

Reply:
I was thinking all the props could be optional in `OptionsData`, but I guess that's the purpose of `$Shape` so we may as well use it.  I was thinking that by having all props in `OptionsData` look like `prop?: T` that would communicate to people that each prop was optional, but it should be pretty obvious by the existing usage.

Reply:
Let's use `$Shape` for now and if people find it confusing we can always change it later.

Reply:
Yeah; it doesn't seem like `style` could be made optional because `Options.havingCrampedStyle` and `Options.havingSize` both use `style` without null checks.